src/conn/conn_evict.c

```diff
@@ -0,0 +1,380 @@
+         * percentage setting and do not allow an absolute size setting.
+         */
+        if (shared)
+            WT_RET_MSG(session, EINVAL,
+              "Shared cache configuration requires a percentage value for %s", param_name);
+        /* An absolute value can't exceed the cache size. */
+        if (input > conn->cache_size)
+            WT_RET_MSG(session, EINVAL, "%s should not exceed cache size", param_name);
+
+        *param = (input * 100.0) / (conn->cache_size);
+    }
+
+    return (0);
+}
+
+/*
+ * __validate_evict_config --
+ *     Validate trigger and target values of given configs.
+ */
+static int
+__validate_evict_config(WT_SESSION_IMPL *session, const char *cfg[])
+{
+    WT_CONFIG_ITEM cval;
+    WT_CONNECTION_IMPL *conn;
+    WT_EVICT *evict;
+    bool shared;
+
+    conn = S2C(session);
+    evict = conn->evict;
+
+    WT_RET(__wt_config_gets_none(session, cfg, "shared_cache.name", &cval));
+    shared = cval.len != 0;
+
+    /* Debug flags are not yet set when this function runs during connection open. Set it now. */
+    WT_RET(__wt_config_gets(session, cfg, "debug_mode.configuration", &cval));
+    if (cval.val)
+        FLD_SET(conn->debug_flags, WT_CONN_DEBUG_CONFIGURATION);
+    else
+        FLD_CLR(conn->debug_flags, WT_CONN_DEBUG_CONFIGURATION);
+
+    WT_RET(__wt_config_gets(session, cfg, "eviction_target", &cval));
+    evict->eviction_target = (double)cval.val;
+    WT_RET(
+      __evict_config_abs_to_pct(session, &(evict->eviction_target), "eviction target", shared));
+
+    WT_RET(__wt_config_gets(session, cfg, "eviction_trigger", &cval));
+    evict->eviction_trigger = (double)cval.val;
+    WT_RET(
+      __evict_config_abs_to_pct(session, &(evict->eviction_trigger), "eviction trigger", shared));
+
+    WT_RET(__wt_config_gets(session, cfg, "eviction_dirty_target", &cval));
+    evict->eviction_dirty_target = (double)cval.val;
+    WT_RET(__evict_config_abs_to_pct(
+      session, &(evict->eviction_dirty_target), "eviction dirty target", shared));
+
+    WT_RET(__wt_config_gets(session, cfg, "eviction_dirty_trigger", &cval));
+    evict->eviction_dirty_trigger = (double)cval.val;
+    WT_RET(__evict_config_abs_to_pct(
+      session, &(evict->eviction_dirty_trigger), "eviction dirty trigger", shared));
+
+    WT_RET(__wt_config_gets(session, cfg, "eviction_updates_target", &cval));
+    evict->eviction_updates_target = (double)cval.val;
+    WT_RET(__evict_config_abs_to_pct(
+      session, &(evict->eviction_updates_target), "eviction updates target", shared));
+
+    WT_RET(__wt_config_gets(session, cfg, "eviction_updates_trigger", &cval));
+    evict->eviction_updates_trigger = (double)cval.val;
+    WT_RET(__evict_config_abs_to_pct(
+      session, &(evict->eviction_updates_trigger), "eviction updates trigger", shared));
+
+    WT_RET(__wt_config_gets(session, cfg, "eviction_checkpoint_target", &cval));
```

Comment:
I realize much of this is code movement from the other file. But please alphabetize the config parsing unless there is an ordering need to do so.